File: solutions/csharp/bob/1/Bob.cs
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Response(string statement) =>
        statement.Trim() switch
        {
            string s when s.EndsWith('?') && IsYelling(s) => "Calm down, I know what I'm doing!",
            string s when s.EndsWith('?') => "Sure.",
            string s when IsYelling(s) => "Whoa, chill out!",
            string s when s == string.Empty => "Fine. Be that way!",
            _ => "Whatever."
        };

    private static bool IsYelling(string s)
    {
        Regex anyLetters = new Regex(@"[a-zA-Z]+");
        return s.ToUpper() == s && anyLetters.Match(s).Success;
    }
}

File: solutions/typescript/bob/5/bob.ts
export function hey(message: string): string {
  message = message.trim()

  if (isLoud(message) && isInquisitive(message))
    return "Calm down, I know what I'm doing!"
  if (isLoud(message))
    return "Whoa, chill out!"
  if (isInquisitive(message))
    return "Sure."
  if (isNothing(message))
    return "Fine. Be that way!"
  return "Whatever."
}

const isLoud = (message: string): boolean =>
  /^.*[A-Z].*$/.test(message) && !/^.*[a-z].*$/.test(message)

const isInquisitive = (message: string): boolean =>
  message.endsWith('?')

const isNothing = (message: string): boolean =>
  message.length === 0
